### DISCUSSION TITLE
Blog on root level supported?

### DISCUSSION THREAD

**Discussion by odin568** [squidfunk/mkdocs-material#6395] [Q&A] [answered]
Hey,
I use mkdocs material for multiple technical documentations. It rocks!
Now I am thinking to add it also for a regular homepage of our fire department as there is not so much content that it deserves a full blown CMS. Dynamic parts are anyhow integrated as iframe.

So what we mainly have is a list of news, where I would use the blog feature.

Now my question: I can follow the docs and create a blog in the "blog" folder, which is then shown as "Blog" as a section in the top bar.
In the sidebar I then have the blog, archive and categories - all fine, see screenshot:

![Bildschirmfoto vom 2023-11-26 19-55-07](https://github.com/squidfunk/mkdocs-material/assets/25813640/067e060a-da75-401c-ba50-40dccf76bc62)

But I would like to have this on the root page. So I moved everything into the docs folder directly (the index.md, the posts folder and the demo post) and configured it with `blog_dir: .`

Now I get the blog on the landing page, but I get the archive and categories in the section on top and not in the sidebar:

![Bildschirmfoto vom 2023-11-26 19-56-09](https://github.com/squidfunk/mkdocs-material/assets/25813640/a9cab6cc-0a35-4393-ab6d-bfb8eb4f9f62)

What am I doing wrong?

**Comment by alexvoss** [comment]
Hi, the [instructions in the blog post announcing the blog plugin](https://squidfunk.github.io/mkdocs-material/blog/2022/09/12/blog-support-just-landed/) give me the arrangement you are looking for. Have a look to see what the differences are in the `mkdocs.yml` that you are using and the one shown there.

**Reply by squidfunk** [reply]
Jup, you just need to remove `navigation.tabs` from `theme.features`

**Comment by odin568** [comment]
Ok, got it thanks!

You are right, removing `navigation.tabs` solved it in a way. 

But then I of course don't have tabs anymore. :smile: So there is no possibility to have the blog with "blog sections" on the left bar on the home but for other folders the usual section behavior on tab bar?

**Reply by squidfunk** [reply]
I don't have the time to test this, but something like the following might work. Keep `navigation.tabs` move the blog to the root, and then specify navigation something like the following:

``` yaml
plugins:
  - blog:
      blog_dir: .
nav:
  - Blog:
    - index.md
  - Non-Blog-Section:
    - ...
  - ...
```

The blog plugin should the attach the categories and archive pages to the `Blog` subsection, and allow you to add additional sections in the navigation tabs. Again, I haven't tested this, but maybe somebody else could try and help.

**Reply by alexvoss** [reply]
I think I might try and turn that into one of our [worked examples](https://github.com/mkdocs-material/examples) today.

**Reply by alexvoss** [reply, marked as answer]
I created a draft PR for the example at https://github.com/mkdocs-material/examples/pull/14. @odin568, if you have a spare moment, can you have a look and check if this does what you imagined? Just 

- clone the [examples repository](https://github.com/mkdocs-material/examples) locally, 
- change into it and `git switch standaloneblog`
- change into `examples/blog-standalone` 
- and run `mkdocs serve`. 
 
The README file is rendered at http://localhost:8000 while the actual example is at http://localhost:8000/demo/.

**Reply by odin568** [reply]
Yeah, works like a charm! 
![Bildschirmfoto vom 2023-11-29 17-48-48](https://github.com/squidfunk/mkdocs-material/assets/25813640/640bf98e-b588-4b8a-b97e-34b31d6d44cc)
It's not a downside for me, but I then have to use the nav section in configs to control it. But this is perfectly fine. In tech documentations always avoided it and simply took the folder names.

Great for your awesome support @squidfunk  and @alexvoss !!

**Reply by odin568** [reply]
So far everything works as said, I just found out one glitch, not sure if it can be resolved.
It only happens, when I'm on the mobile phone. When I open the burger menu, I see the blog sections if I open the homepage with the blog on the root:
![grafik](https://github.com/user-attachments/assets/0c726b1c-f8ab-4818-b71b-41c1ea6a11dc)

Ideally, I would see the sections in thise case:
![grafik](https://github.com/user-attachments/assets/97a6f4ac-bfbc-4942-aef2-8627ac95bb96)

You may have a look also on yourself: https://ffw-baudenbach.de

Is there anything I can do? It might be misleading if opening the homepage from mobile phone that you first need to first click the "back" arrow to see the "regular menu".

**Reply by squidfunk** [reply]
This is working as intended. The mobile navigation currently works in layers, and when you open it, the active layer is on top, similar to how the active section is expanded on desktop. I'm actually thinking about restructuring the mobile navigation to work more like the desktop one, but we currently have other priorities, so it's a "likely in the future" for now.

**Comment by alexvoss** [comment]
The answer with Material for MkDocs is rarely that you can't. I don't think what you are looking for can be done with changes to the `mkdocs.yml` alone. You would need to replace the left-hand side navigation on the homepage with the blog navigation. You can do this by [overriding](https://squidfunk.github.io/mkdocs-material/customization/?h=block#overriding-blocks) the `site_nav` block.
